grpc/execution/server.go

```diff
@@ -114,6 +115,16 @@ func (s *ExecutionServiceServer) DoBlock(ctx context.Context, req *executionv1.D
 	return res, nil
 }
 
+func (s *ExecutionServiceServer) FinalizeBlock(ctx context.Context, req *executionv1.FinalizeBlockRequest) (*executionv1.FinalizeBlockResponse, error) {
+	header := s.bc.GetHeaderByHash(common.BytesToHash(req.BlockHash))
+	if header == nil {
```

Comment:
Should we also be validating that this is a part of the canonical chain?

Reply:
what do you mean? if we have a non-canonical fork, and we finalize a block on it, it becomes canonical. finalization overrides non-finalized but otherwise canonical chains

Reply:
also in our case, we don't have forks anyways